platform/android/src/style/android_conversion.hpp

```diff
@@ -110,11 +110,32 @@ class ConversionTraits<mbgl::android::Value> {
     }
 
     static optional<GeoJSON> toGeoJSON(const mbgl::android::Value &value, Error &error) {
-        if (value.isNull() || !value.isString()) {
+        if (value.isNull()) {
             error = { "no json data found" };
             return {};
         }
-        return parseGeoJSON(value.toString(), error);
+
+        if (value.isString()) {
+            return parseGeoJSON(value.toString(), error);
+        }
+
+        if (value.isObject()) {
+            mbgl::android::Value keys = value.keyArray();
+            std::size_t length = arrayLength(keys);
+            for (std::size_t i = 0; i < length; ++i) {
+                const auto k = keys.get(i).toString();
+                if (k == "json") {
+                    auto v = value.get("json");
+                    if (v.isString()) {
+                        return parseGeoJSON(v.toString(), error);
+                    } else {
```

Comment:
nit: else is not needed

Reply:
If v is not string that means the json format is not correct. In this case, there is no need to continue iterate the map.